packages/repository/src/repository-mixin.ts

```diff
@@ -0,0 +1,127 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import { Class } from './common-types';
+import { MixinBuilder } from './mixin';
+import { Component } from '@loopback/core';
+import { Repository } from './repository';
+
+// tslint:disable:no-any
+
+/**
+* A mixin class for Application that creates a .repository()
+* function to register a repository automatically. Also overrides
+* component function to allow it to register repositories automatically.
+*
+* ```ts
+*
+* class MyApplication extends RepositoryMixin(Application) {}
+* ```
+*/
+export function RepositoryMixin<T extends Class<any>>(superClass: T) {
+  return class extends superClass {
+    // A mixin class has to take in a type any[] argument!
+    constructor(...args: any[]) {
```

Comment:
I think the requirement for the constructor to take in any arguments may not be caused by mixin constraints, but because the `Class` is defined with that constructor!

See https://github.com/strongloop/loopback-next/blob/e8721422e9a6ea3535e5da75691c464a8fa5cfa5/packages/repository/src/common-types.ts#L14-L19

Reply:
As per TypeScript Docs (and the fact that changing the `..args: any[]` throws an error), a Mixin must take in any arguments. 

From TS Docs: 
> The constructor of a mixin class (if any) must have a single rest parameter of type any[] and must use the spread operator to pass those parameters as arguments in a super(...args) call.

Reference (The 2 bullet points): http://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-2.html

Reply:
Interesting, I didn't know about that. Thank you for educating me :)